pkg/controllers/project/controller.go

```diff
@@ -151,10 +151,15 @@ func (h *handler) shouldManage(projectHelmChart *v1alpha1.ProjectHelmChart) (boo
 	if projectHelmChart == nil {
 		return false, nil
 	}
-	isProjectRegistrationNamespace, err := h.projectGetter.IsProjectRegistrationNamespace(projectHelmChart.Namespace)
+	namespace, err := h.namespaceCache.Get(projectHelmChart.Namespace)
 	if err != nil {
-		return false, err
+		// If the namespace that the projectHelmChart resides in does not exist, it shouldn't be managed
+		//
+		// Note: we know that this error would only happen if the namespace is not found since the only valid error returned from this
+		// call is errors.NewNotFound(c.resource, name)
```

Comment:
There are only two possible errors here:

https://github.com/rancher/helm-project-operator/blob/ecbf39ae3427ad2fe1704cd8bbd4556b3a2fc4ce/pkg/generated/controllers/helm.cattle.io/v1alpha1/projecthelmchart.go#L226-L235

The indexer is defined here:

https://github.com/rancher/helm-project-operator/blob/ecbf39ae3427ad2fe1704cd8bbd4556b3a2fc4ce/pkg/generated/controllers/helm.cattle.io/v1alpha1/projecthelmchart.go#L175

This indexer will always be a [`cache.cache`](https://github.com/kubernetes/client-go/blob/704cac462081473b521f15547302f001f78ef2e4/tools/cache/store.go#L234-L239) that uses a [`cache.ThreadSafeStore`](https://pkg.go.dev/k8s.io/client-go/tools/cache#ThreadSafeStore) as the `cacheStorage`, and this **always** returns nil for the error.

Therefore, the only possible error is that it does not exist.